inginious/frontend/pages/course_admin/submissions_data.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+from flask import request
+from bson.json_util import dumps
+
+from inginious.frontend.pages.course_admin.api_auth import DataAPIPage
+
+class SubmissionsEndpoint(DataAPIPage):
+    """ Endpoint to retrieve submission data """
+
+    def GET(self, courseid):
+        """ GET request """
+        courseid = self.verify()
+
+        # DB request parameters
+        params = {"courseid": courseid}
```

Comment:
It seems you don't check the token course scope and give access to the whole database as long as the token is valid.